src/__test__/components/data-exploration/heatmap/HeatmapPlot.test.jsx

```diff
@@ -1,545 +1,342 @@
-        children: louvainClusters,
-      }]));
-
-    component = mount(
-      <Provider store={store}>
-        <HeatmapPlot experimentId={experimentId} width={200} height={200} />
-      </Provider>,
-    );
+    // It doesn't show the cells for louvain-3 anymore
+    expect(isSubset(cellsInLouvain3, vitesscePropsSpy.expressionMatrix.rows)).toEqual(false);
+
+    // Keeps all the other cells and genes the same
+    expect(vitesscePropsSpy.expressionMatrix.rows).toMatchSnapshot();
+    expect(vitesscePropsSpy.expressionMatrix.cols).toMatchSnapshot();
 
-    expect(component.find('HeatmapPlot').length).toEqual(1);
+    // If a louvain-3 is shown again
+    await act(async () => {
+      storeState.dispatch(setCellSetHiddenStatus(experimentId, 'louvain-3'));
+    });
 
-    const expectedMarkerGenes = 3;
+    // It shows the cells for louvain-3 again
+    expect(isSubset(cellsInLouvain3, vitesscePropsSpy.expressionMatrix.rows)).toEqual(true);
 
-    expect(loadMarkerGenes).toHaveBeenCalledWith('123', 10, 'interactiveHeatmap', expectedMarkerGenes);
+    // Keeps all the other cells and genes the same
+    expect(vitesscePropsSpy.expressionMatrix.rows).toMatchSnapshot();
+    expect(vitesscePropsSpy.expressionMatrix.cols).toMatchSnapshot();
   });
 
-  it('doesn\'t load marker genes if it doesn\'t know the louvain clusters', async () => {
-    const store = mockStore({
-      ...initialState,
-      networkResources: { environment: Environment.DEVELOPMENT },
-      cellSets: {
-        ...initialState.cellSets,
-        loading: true,
-        error: false,
-      },
-      genes: {
-        ...initialState.genes,
-        expression: {
-          ...initialState.genes.expression,
-          loading: false,
-          error: 'Some error',
-        },
-      },
-      experimentSettings: {
-        ...initialState.experimentSettings,
-        processing: {
-          ...initialState.processing,
-          configureEmbedding: {
-            clusteringSettings: { methodSettings: { louvain: { resolution: 10 } } },
-          },
-        },
-      },
-    });
+  it('Reacts to cellClass groupby being changed', async () => {
+    fetchMock.mockIf(/.*/, mockAPI(generateDefaultMockAPIResponses(experimentId, fake.PROJECT_ID)));
 
-    getCellSetsHierarchyByKeys.mockReturnValue(() => []);
+    seekFromAPI.mockImplementation((a, b, c, requested) => mockWorkerResponses[requested]());
 
-    component = mount(
-      <Provider store={store}>
-        <HeatmapPlot experimentId={experimentId} width={200} height={200} />
-      </Provider>,
-    );
+    await loadAndRenderDefaultHeatmap();
+
+    // Renders correctly
+    expect(screen.getByText(/Sup Im a heatmap/i)).toBeInTheDocument();
 
-    expect(component.find('HeatmapPlot').length).toEqual(1);
+    // If groupedTracks change
+    await act(async () => {
+      storeState.dispatch(
+        updatePlotConfig('interactiveHeatmap', {
+          groupedTracks: ['sample', 'louvain'],
+        }),
+      );
```

Comment:
was there not an easy way to trigger change of groupedTracks via the ui rather than having to dispatch an action?

Reply:
No, because we are rendering the heatmapplot only, the groupedTracks are changed in a different component